pinot-core/src/main/java/org/apache/pinot/core/common/Operator.java

```diff
@@ -39,12 +40,20 @@
    */
   T nextBlock();
 
+  /** @return Name of this operator */
+  String getOperatorName();
+
+  /** @return List of {@link Operator}s that this operator depends upon. */
+  List<Operator> getChildOperators();
+
   /**
-   * Returns the name of the operator.
-   * NOTE: This method is called for tracing purpose. The sub-class should try to return a constant to avoid the
-   * unnecessary overhead.
+   * @return EXPLAIN PLAN name if one is defined; otherwise, null. null EXPLAIN PLAN name means that this operator
+   * won't be included in EXPLAIN PLAN output.
    */
-  String getOperatorName();
+  String getExplainPlanName();
```

Comment:
I think we should be able to merge this with `toExplainString()`, and annotate it with nullable. Don't see the value of keeping 2 separate methods

Reply:
^^